Framework/API/src/MatrixWorkspace.cpp

```diff
@@ -2047,6 +2055,80 @@ IPropertyManager::getValue<Mantid::API::MatrixWorkspace_const_sptr>(
   }
 }
 
+template <>
```

Comment:
Hmm, looks tricky, basically we would need to convert a `std::tuple<T1 &, T2&>` to `std::tuple<T1, T2>`. I will think some more about it, probably it could be managable with some template magic.